## Uploading and accessing data

* [How to register to provide metadata and/or sequence data](register)
* [How to upload sequences to CLIMB](upload-instructions)
* [What is the timeline for the automated CLIMB pipelines?](when-metadata)
* [What happens to data uploaded to CLIMB](how-data)
* [How to access data stored on CLIMB](accessing-data)
* [How to access a dataview using the API](accessing-dataviews)

## Providing metadata

* [Basic terminology and minimal requirements](metadata)
* [You think you are ready to submit metadata?](metadata-tips)
* [Cycle threshold validators](https://docs.google.com/spreadsheets/d/1UWGvjfj-u0g5-N_DtYwqyGTrS5duOOqwg3jr28Y4hLw/edit?usp=sharing)
* [Validating a Heron barcode](heron-barcodes)
* [Uploading a library and sequencing run without the biosample](uploading-nosample)
* [Using investigation fields to label specific sample types](investigations)

## Outbound distribution

* [How to opt-in to automated GISAID submissions](gisaid)

## Infrastructure

* [Authorising applications to use your Majora account using OAuth](oauth)
* [Registering and using an OAuth application](oauth-app)
* [Joining and using the interpipeline communucation network](tael)
* [Majora API documentation for developers and advanced users](https://redocly.github.io/redoc/?url=http://majora.covid19.climb.ac.uk/static/majora.yaml)

## Lab protocols

* [COG-UK sequencing protocols group on Protocols.io](https://protocols.io/groups/coguk)
* [WSI COVID-19 ARTIC v3 Illumina library construction and sequencing protocol (version:bgq3jvyn)](https://dx.doi.org/10.17504/protocols.io.bgq3jvyn)

## Bioinformatics pipelines

* [Nextflow workflows for amplicon Illumina and nanopore](https://github.com/connor-lab/ncov2019-artic-nf)

## CoV-GLUE

* [Accessing CoV-GLUE on CLIMB](cov-glue/climb-glue.md)
* [Reporting primer and probe mismatches using CoV-GLUE](cov-glue/cov-glue-primers-reporting.md)

## Meta
* [Adding a page to this site](how-to-page)
